ludwig/utils/trainer_utils.py

```diff
@@ -408,3 +410,16 @@ def get_rendered_batch_size_grad_accum(config: "BaseTrainerConfig", num_workers:
                 gradient_accumulation_steps = 1
 
     return batch_size, gradient_accumulation_steps
+
+
+def freeze_layers_regex(config: "BaseTrainerConfig", model: ECD) -> None:
+    """Freezes layers based on provided regular expression."""
```

Comment:
lets add all of the comments around inputs/outputs as well

Reply:
@ethanreidel I think that if you put `rom __future__ import annotations` as the very first line in the module, you would not need to quote the types.  Would you like to give it a try and see if it works?  Thanks!

Reply:
Fixed. Thanks

Reply:
I tested the annotations import, and it worked, but the git pre-commit was forcing changes (e.g. converting all uppercase Dicts to lowercase dicts) that I didn't like.

Reply:
@ethanreidel I think that makes sense. 

Are you able to expand on the docstring itself for this function? Also, if it also supports `LLM`, can we make `model` a union of ECD and LLM?